core/system/inc/page_allocator_config.h

```diff
@@ -25,21 +25,88 @@
  * a relatively low limit to the number of pages.
  * By default a maximum of 16 pages are allowed. This can only be overwritten
  * by the porting engineer for the current platform. */
-#ifndef UVISOR_PAGE_TABLE_MAX_COUNT
-#define UVISOR_PAGE_TABLE_MAX_COUNT ((uint32_t) 16)
+#ifndef UVISOR_PAGE_MAX_COUNT
+#define UVISOR_PAGE_MAX_COUNT (16UL)
 #endif
 /* The number of pages is decided by the page size. A small page size leads to
  * a lot of pages, however, number of pages is capped for efficiency.
  * Furthermore, when allocating large continous memory, a too small page size
  * will lead to allocation failures. This can only be overwritten
  * by the porting engineer for the current platform. */
 #ifndef UVISOR_PAGE_SIZE_MINIMUM
-#define UVISOR_PAGE_SIZE_MINIMUM ((uint32_t) 1024)
+#define UVISOR_PAGE_SIZE_MINIMUM (1024UL)
 #endif
+/* The page owner map internally uses a right-aligned bit mask, with the LSB
+ * corresponding to the lowest page index and the MSB to the highest index.
+ *
+ * For ARMv7-M MPU implementation, the page heap is protected using subregions.
+ * Since the end of the page heap is aligned with the corresponding region size,
+ * we require accessing the upper page owner mask bits in 8-bit segments.
+ * This is significantly easier and much more efficient to do if the page owner
+ * map is in left-aligned format. */
+#ifdef ARCH_MPU_ARMv7M
+#define UVISOR_PAGE_OWNER_MAP_LEFT_ALIGNED 1
+#else
+#define UVISOR_PAGE_OWNER_MAP_LEFT_ALIGNED 0
+#endif
+/* Defines the number of uint32_t page owner masks in the owner map. */
+#define UVISOR_PAGE_MAP_COUNT ((UVISOR_PAGE_MAX_COUNT + 31) / 32)
 
 /* The page box_id is the box id which is 8-bit large. */
 typedef uint8_t page_owner_t;
 /* Define a unused value for the page table. */
 #define UVISOR_PAGE_UNUSED ((page_owner_t) (-1))
+/* Contains the total number of available pages. */
+extern uint8_t g_page_count_total;
+
+/** Sets the page bit in the page map array.
+ * @param map   an array of `uint32_t` containing the page map
+ * @param page  the index of the page to be set
+ */
+static inline void page_allocator_map_set(uint32_t * const map, uint8_t page)
+{
+#if UVISOR_PAGE_OWNER_MAP_LEFT_ALIGNED
+    page += UVISOR_PAGE_MAP_COUNT * 32 - g_page_count_total;
+#endif
+#if UVISOR_PAGE_MAP_COUNT == 1
```

Comment:
@niklas-arm why sis there a special case for UVISOR_PAGE_MAP_COUNT == 1 ? Isn't that covered by the general case below?